src/View.elm

```diff
@@ -0,0 +1,190 @@
+import Model exposing (..)
+import RemoteData exposing (RemoteData(..))
+
+
+viewStyled : Model -> Html Msg
+viewStyled model =
+    div [] [ viewRemoteData "list of users" model.users (viewUsers model) ]
+
+
+viewRemoteData : String -> RemoteData (List String) a -> (a -> Html Msg) -> Html Msg
+viewRemoteData elementName rdata viewSuccess =
+    case rdata of
+        NotAsked ->
+            text ("Initializing " ++ elementName)
+
+        Loading ->
+            text ("Loading " ++ elementName)
+
+        Failure errors ->
+            text ("Error loading " ++ elementName ++ ": " ++ String.join ", " errors)
+
+        Success x ->
+            viewSuccess x
+
+
+viewUsers : Model -> Dict Int User -> Html Msg
+viewUsers ({ underEdit } as model) users =
+    let
+        theUsers =
+            Dict.toList users
+
+        medallions =
+            (List.take 3 theUsers |> List.map (viewUser medallion underEdit)) ++ [ viewCreateUserMedallion underEdit ]
+
+        medallionsArea =
+            div
+                [ css
+                    [ property "display" "grid"
+                    , property "grid-template-columns" "repeat(auto-fill, minmax(min(100%, 180px), 1fr))"
+                    , property "grid-gap" "8px"
+                    , padding (px 8)
+                    , borderBottom2 (px 1) solid
+                    ]
+                ]
+                medallions
+
+        rows =
+            List.drop 3 theUsers |> List.map (viewUser row underEdit)
+    in
+    div [] (viewFooter model :: medallionsArea :: rows)
+
+
+viewFooter : Model -> Html Msg
+viewFooter { confirmingDelete, errors, underEdit } =
+    case ( confirmingDelete, errors, underEdit ) of
+        ( Just ( id, { firstName, lastName } ), _, _ ) ->
+            modalFooter [ css [ backgroundColor (hex "eaa") ] ]
+                [ text ("Do you want to delete " ++ firstName ++ " " ++ lastName ++ "? ")
+                , button [ onClick (DeleteUser id) ] [ text "yes" ]
+                , button [ onClick (IntendDeleteUser Nothing) ] [ text "no" ]
+                ]
+
+        ( Nothing, errs :: _, _ ) ->
+            footer [ css [ backgroundColor (hex "eaa") ] ]
+                [ text ("Error: " ++ String.join ", " errs)
+                , button [ onClick DismissLastError ] [ text "dismiss" ]
+                ]
+
+        ( Nothing, [], NewUser user ) ->
+            footer [ css [ backgroundColor (hex "aea") ] ]
+                [ button [ onClick (PostUser user) ] [ text "save" ]
+                , button [ onClick (Edit NotEditing) ] [ text "discard" ]
+                ]
+
+        ( Nothing, [], Id id user ) ->
+            footer [ css [ backgroundColor (hex "eea") ] ]
+                [ button [ onClick (PutUser id user) ] [ text "save" ]
+                , button [ onClick (Edit NotEditing) ] [ text "discard" ]
+                , button [ onClick (IntendDeleteUser (Just ( id, user ))) ] [ text "delete" ]
+                ]
+
+        _ ->
```

Comment:
So far I like your code, this is the first time I was triggered. I know it looks OK now, to use wildcard, but for future changes it's always beneficial to avoid them as much as possible. This is one of those rules which pay of.

Reply:
Thanks for your input! You're right, instead of the default branch there's only one possibility left, which is better to spell out explicitly.